src/env/node.js

```diff
@@ -12,13 +12,26 @@ import { globSync } from 'glob';
 import format from "../format-console.js";
 import { getType, interceptConsole, restoreConsole } from '../util.js';
 
-// Recursively traverse a subtree starting from `node` and make (only) groups of tests collapsible
-function makeCollapsible (node) {
+/**
+ * Recursively traverse a subtree starting from `node` and make (only) groups of tests collapsible.
+ * @param {object} node
+ * @param {string} [options.expand = "fail"] Space-separated list of possible test results: `pass`, `fail`, or `skipped`.
+ *        Nodes (groups of tests) that contain tests with either of the specified results will be expanded.
+ * @param {boolean} [options.verbose] Show all tests, not just failed ones.
+ * @example makeCollapsible(root, { expand: "fail skipped", verbose: true });
+ */
+function makeCollapsible (node, options = {}) {
 	if (node.tests?.length) {
-		node.collapsed = true; // all groups are collapsed by default
+		let expand = [...new Set((options.expand ?? "fail").split(/\s+/))];
+
+		// Root is collapsed by default.
+		// All groups are collapsed by default or if the verbose option is specified.
+		// Otherwise, it will be expanded if a group contains tests, because of which
+		// the group should be expanded (e.g., failed tests).
+		node.collapsed = node.parent && !options.verbose && expand.some(o => node.stats[o] > 0) ? false : true;
```

Comment:
I’m not quite sure about this `expand` setting. What other use cases does `expand` have besides fail? How would you disable the auto-expansion for fail?

Also, perhaps you could split that into a separate PR so we can merge the formatting changes asap.